### ISSUE TITLE
[BUG] value "todie" is set to 99999 when 100% efficiency is met

### ISSUE THREAD

**Issue by rakasunka** [deurk/ktx#100]
**Describe the bug**
The value todie is set to 99999 when 100% efficiency is met

**To Reproduce**
Win a game without dying.

**Expected behavior**
No idea what value of todie should be when a game is played without a death.

**Screenshots**
https://imgur.com/a/G7cxJML
[100eff.zip](https://github.com/deurk/ktx/files/4566920/100eff.zip) <- mvd

**Environment(please complete the following information):**
 - Operation System: Linux
 - Processor Architecture: unknown
 - MVDSV Version: 0.32
 - KTX Version: 1.38

**Additional context**
Game was played on getquad official server.

**Comment by deurk** [comment]
Well, 0 wouldn't make any sense, 99999 is just a silly value as well. What do you propose we put?
Just "todie: N/A"?

**Comment by meag** [comment]
match_duration * 60?

**Comment by deurk** [comment]
I thought _todie_ was total damage taken divided by number of deaths?

**Comment by meag** [comment]
Ah sorry, I thought it was time-based... although that wouldn't work either.

Can we change it to be total damage taken divided by (number of deaths + 1)?  So it's average damage taken per life?

**Comment by deurk** [comment]
Well, you're not technically dead at the end if you have any health. Should we remove health left and just divide by deaths? No deaths would mean _todie_ = _damage_taken_

**Comment by meag** [comment]
Think this is one for @andrestone ...

**Comment by andrestone** [comment]
I think it would be nice to display a distinguishable information. Total damage taken would be my second choice.

**Comment by deurk** [comment]
"undead"? 😄

**Comment by andrestone** [comment]
Setting like that... How about NaN? Since it would be a division by zero? We don't have infinity symbol in the charset, do we?

**Comment by deurk** [comment]
Well, if we take the extreme case where several players end up with no death, taken damage will be the way to see who actually deserves the survivor trophy, so I think with no death, displayed value should be _damage_taken - health_left_

**Comment by mushis** [comment]
there's one more possible bug on the stats: did you took 1 rl during the whole game?

**Comment by rakasunka** [comment]
I didn't take any RL at all during the game, blood dog dropped rl packs.. there was maybe 7-8 rl packs dropped that i took..

**Comment by deurk** [comment]
> Well, if we take the extreme case where several players end up with no death, taken damage will be the way to see who actually deserves the survivor trophy, so I think with no death, displayed value should be _damage_taken - health_left_

Actually _damage_taken + health_left_

**Comment by deurk** [comment]
Realized calculation was wrong.

If player died at least once: _todie = damage taken / number of deaths_ (we're not tracking damage taken during the last life if not dead at end of match though so value is a bit higher than it should)
If player did not die: _todie = damage taken + remaining health + remaining armor_ (because it would have taken at least this amount of damage to kill the player and that's all we can guess)

**Comment by andrestone** [comment]
Yeah, if we want to consider a never dead player as someone who is dead at
the end of the game. The armor + health math isn't that straight forward,
though. 150 RA + 10h isn't the same as 150YA + 10h, for example.

**Comment by meag** [comment]
Good news: there's a function the bots use that will give you that.

`TotalStrength (client->s.v.health, client->s.v.armorvalue, client->s.v.armortype)`

Bad news: it'll return 1000 if the match is already over (match_in_progress), and it'll need moved out of the `BOT_SUPPORT` section

**Comment by andrestone** [comment]
Copy/pasta FTW!
